test/test_plugin_threema.py

```diff
@@ -0,0 +1,127 @@
+# Copyright (c) 2023, Chris Caron <lead2gold@gmail.com>
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# 1. Redistributions of source code must retain the above copyright notice,
+#    this list of conditions and the following disclaimer.
+#
+# 2. Redistributions in binary form must reproduce the above copyright notice,
+#    this list of conditions and the following disclaimer in the documentation
+#    and/or other materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+
+from unittest import mock
+
+import pytest
+import requests
+
+from apprise.plugins.NotifyThreema import NotifyThreema
+from helpers import AppriseURLTester
+
+# Disable logging for a cleaner testing output
+import logging
+logging.disable(logging.CRITICAL)
+
+# Our Testing URLs
+apprise_url_tests = (
+    ('threema://', {
+        # No user/secret specified
+        'instance': TypeError,
+    }),
+    ('threema://@:', {
+        # Invalid url
+        'instance': TypeError,
+    }),
+    ('threema://user@secret', {
+        # username must be 8 characters in len
+        'instance': TypeError,
+    }),
+    ('threema://username@secret/{targets}/'.format(
+        targets='/'.join(['2222'])), {
+
+        # Invalid target phone number
+        'instance': NotifyThreema,
+        'notify_response': False,
+        'privacy_url': 'threema://u...e@s...t/2222',
+    }),
+    ('threema://username@secret/{targets}/'.format(
+        targets='/'.join(['16134442222'])), {
+
+        # Valid
+        'instance': NotifyThreema,
+        'privacy_url': 'threema://u...e@s...t/16134442222',
+    }),
+    ('threema://username@secret/{targets}/'.format(
+        targets='/'.join(['16134442222', '16134443333'])), {
+
+        # Valid multiple targets
+        'instance': NotifyThreema,
+        'privacy_url': 'threema://u...e@s...t/16134442222/16134443333',
+    }),
+    ('threema:///?secret=secret&from=username&to={targets}'.format(
+        targets=','.join(['16134448888', 'user@gmail.com', 'abcd1234'])), {
+
+        # Valid
+        'instance': NotifyThreema,
+    }),
+    ('threema://username@secret/', {
```

Comment:
Also what is that `secret` here? Cause [Threema has it's own syntax for that on clients](https://threema.ch/en/faq/url_actions_old), [though the URL syntax is preferred nowadays](https://threema.ch/en/faq/url_actions)

But ahh, now I get it, this is an Apprise internal thing? So not related to Threema URIs or the clients, I see.

Reply:
No, it's not an Apprise internal thing.  I thought the  `secret` was part of the Threema Gateway API (Simple Mode) Payload?  In order to pass it through to your upstream server, Apprise needs to be aware of it.  Or am i misinterpreting something?

Reply:
NO Imeant the URI action in general i.e. `threema://`  sorry for the confusion. Because on Android you can also open a `threema://ThreemaId`  link and it opens in Threema. See the links I provided.

Reply:
My thoughts are: different program, different use-case. I support `xml://`, `json://` and even parse `https://` which no doubt could clash with other systems that leverage the URI approach too. See the `slack://` or `discord://` wiki and you can still trigger a notification to those services if you just paste the webhook served directly to you from their admin area (`https://`). 🙂 Hopefully that makes sense for the reasoning.